roles/backpack_kube/tasks/main.yml

```diff
@@ -1,48 +1,79 @@
+
 - name: Create Backpack Namespace
   k8s: 
-    name: backpack
+    name: "{{ operator_namespace }}"
     api_version: v1
     kind: Namespace
     state: present
     kubeconfig: "{{ kube_config }}"
 
-- name: Launch backpack daemonset
+- name: Convert kubeconfig to base64
+  shell: "base64 {{ kube_config }} | tr -d '\n'"
+  register: metadata_kubeconfig
+
+- name: Create UUID
+  command: "uuidgen"
+  register: my_uuid
+
+- name: Set UUID
+  set_fact:
+    uuid: "{{ my_uuid.stdout_lines|first }}"
+
+- name: Generate trunc_uuid
+  shell: "echo {{ uuid }} | awk -F '-' '{print $1}'"
+  register: my_trunc_uuid
+
+- name: Set trunc_uuid
+  set_fact:
+    trunc_uuid: "{{ my_trunc_uuid.stdout_lines|first }}"
+
+- name: Create kubeconfig secret
   k8s:
     state: present
-    definition: "{{ lookup('file', 'roles/backpack_kube/files/backpack_set.yaml') }}"
+    definition: "{{ lookup('template', 'backpack_secret.yml') | from_yaml }}"
     kubeconfig: "{{ kube_config }}"
-  register: my_daemon_set
 
-- name: Sleep 5 to allow process time
-  pause:
-    seconds: 5
-
-- name: Wait until all pods exit creating state
-  k8s_facts:
-    kind: Pod
+- name: Create backpack DaemonSet
+  k8s:
+    state: present
+    definition: "{{ lookup('template', 'backpack.yml') | from_yaml }}"
     kubeconfig: "{{ kube_config }}"
-    label_selectors:
-      - name = backpack
-  register: pod_list
-  until: "'Creating' not in pod_list | json_query('resources[].status.phase')"
+  register: my_daemonset
 
-
-- name: Wait for all pods to contain BACKPACK_DONE
-  shell: "/usr/local/bin/kubectl --kubeconfig={{ kube_config }} -n backpack logs -l name=backpack | grep BACKPACK_DONE | wc -l"
-  register: done_count
-  until: "done_count.stdout|int == pod_list.resources|length"
-  retries: 60
-  delay: 10
+- name: Wait for daemonset to be ready
+  k8s_facts:
+    api_version: apps/v1
+    kind: DaemonSet
+    name: "backpack-{{ trunc_uuid }}"
+    namespace: "{{ operator_namespace }}"
+  register: my_daemonset
+  until: "my_daemonset | json_query('resources[].status.desiredNumberScheduled') == my_daemonset | json_query('resources[].status.numberReady') and my_daemonset | json_query('resources[].status.numberReady') != 0 "
+  retries: 120
+  delay: 5
 
 - name: Create backpack output directory
   file:
     path: "{{ backpack_output_dir }}"
     state: directory
     mode: "u=rw,g=rwx,o=rwx"
 
+- name: Get poid list
+  k8s_facts:
+    kind: Pod
+    namespace: "{{ operator_namespace }}"
+    label_selectors:
+      - name = backpack-{{ trunc_uuid }}
+  register: pod_list
+
 - name: Dump logs to temporary directory
-  shell: "/usr/local/bin/kubectl --kubeconfig={{ kube_config }} -n backpack logs {{ item }} | grep -v BACKPACK_DONE > {{ backpack_output_dir }}/{{ item }}"
+  shell: "/usr/local/bin/kubectl --kubeconfig={{ kube_config }} -n {{ operator_namespace }} cp {{ item }}:/tmp/stockpile.json {{ backpack_output_dir }}/{{ item }}"
```

Comment:
This works for now, but I hope we can move towards using a k8s ansible module for this ( asked on ansible channel for an answer )

Reply:
Agreed. I'd like to use k8s_copy (if such existed) but I haven't found a better option than this atm.